### ISSUE TITLE
Errors not displayed in text editor for `[type-var]` errors

### ISSUE THREAD

**Issue by onlywei** [microsoft/vscode-mypy#370]
The Mypy extension correctly runs and identifies a [type-var] error, and the error message from the mypy command is correctly logged in the "Mypy Type Checker" output panel. However, the extension then fails to display any diagnostic information (e.g., "red squiggles") in the editor or add an entry to the "Problems" panel.
The verbose logs clearly show the extension receiving the error from mypy, but it immediately sends a textDocument/publishDiagnostics notification with an empty diagnostics array, indicating that the extension's internal parser is likely failing to interpret this specific error format.

This issue occurs with both a project-installed mypy (via Poetry) and the mypy version bundled with the extension.

## Diagnostic Data
- Python version (& distribution if applicable, e.g., Anaconda): 3.12
- Type of virtual environment used (e.g., conda, venv, virtualenv, etc.): Poetry
- Operating system (and version): macOS 15.5
- Version of tool extension you are using: ms-python.mypy-type-checker v2024.2.0 (Universal)

### Expected Behavior
When a `[type-var]` error is detected by mypy, a red squiggle should appear under the incorrect code in the editor, and the error should be listed in the "Problems" panel.

### Actual Behavior
The error text is correctly printed in the "Mypy Type Checker" output panel, but no diagnostics are shown in the editor. The "Problems" panel remains empty.

## Reproduction Steps:

1. In a separate library that is installed as a dependency, define a class that uses a generic TypeVar with a bound. The library must include a `py.typed` file.

```python
    # In a library module (e.g., my_library/models.py)
    from typing import TypeVar, Generic
    class MyModel: pass
    T = TypeVar("T", bound=MyModel)
    class Metadata(Generic[T]):
        def __init__(self, model: type[T], table_name: str | None): ...
```

2. In your project code, import and instantiate that class, but use a type that violates the TypeVar's bound.

```python
    # In your project file (e.g., main.py)
    from my_library.models import Metadata
    class NotAModel: pass
    # This line should cause a [type-var] error
    metadata = Metadata(model=NotAModel, table_name="test")
```

3. Save the file to trigger the Mypy check.
4. Observe the "Mypy Type Checker" output panel and the editor. The error will appear in the output log, but not as a diagnostic in the editor.

## Logs:

<details>
  <summary>Click here for detailed logs</summary>
```
2025-07-14 15:47:54.272 [info] [Trace - 3:47:54 PM] Sending notification 'textDocument/didSave'.
2025-07-14 15:47:54.272 [info] Params: {
    "textDocument": {
        "uri": "file:///Users/weiwang/repos/gitlab-maven/homepage-api/src/homepageapi/repository/content_management/condition.py"
    }
}


2025-07-14 15:47:54.279 [info] [Trace - 3:47:54 PM] Received notification 'window/logMessage'.
2025-07-14 15:47:54.279 [info] Params: {
    "type": 4,
    "message": "/Users/weiwang/repos/gitlab-maven/homepage-api/.venv/bin/python -m mypy --no-color-output --no-error-summary --show-absolute-path --show-column-numbers --show-error-codes --no-pretty --show-error-end /Users/weiwang/repos/gitlab-maven/homepage-api/src/homepageapi/repository/content_management/condition.py"
}


2025-07-14 15:47:54.279 [info] /Users/weiwang/repos/gitlab-maven/homepage-api/.venv/bin/python -m mypy --no-color-output --no-error-summary --show-absolute-path --show-column-numbers --show-error-codes --no-pretty --show-error-end /Users/weiwang/repos/gitlab-maven/homepage-api/src/homepageapi/repository/content_management/condition.py
2025-07-14 15:47:54.279 [info] [Trace - 3:47:54 PM] Received notification 'window/logMessage'.
2025-07-14 15:47:54.279 [info] Params: {
    "type": 4,
    "message": "CWD Server: /Users/weiwang/repos/gitlab-maven/homepage-api"
}


2025-07-14 15:47:54.279 [info] CWD Server: /Users/weiwang/repos/gitlab-maven/homepage-api
2025-07-14 15:47:55.070 [info] [Trace - 3:47:55 PM] Received notification 'window/logMessage'.
2025-07-14 15:47:55.070 [info] Params: {
    "type": 4,
    "message": "/Users/weiwang/repos/gitlab-maven/homepage-api/src/homepageapi/repository/content_management/condition.py:14:16:19:5: error: Value of type variable \"InstanceT\" of \"Metadata\" cannot be \"Condition\"  [type-var]\n"
}


2025-07-14 15:47:55.070 [info] /Users/weiwang/repos/gitlab-maven/homepage-api/src/homepageapi/repository/content_management/condition.py:14:16:19:5: error: Value of type variable "InstanceT" of "Metadata" cannot be "Condition"  [type-var]

2025-07-14 15:47:55.070 [info] [Trace - 3:47:55 PM] Received notification 'window/logMessage'.
2025-07-14 15:47:55.070 [info] Params: {
    "type": 4,
    "message": "file:///Users/weiwang/repos/gitlab-maven/homepage-api/src/homepageapi/repository/content_management/condition.py :\r\n"
}


2025-07-14 15:47:55.070 [info] file:///Users/weiwang/repos/gitlab-maven/homepage-api/src/homepageapi/repository/content_management/condition.py :

2025-07-14 15:47:55.070 [info] [Trace - 3:47:55 PM] Received notification 'textDocument/publishDiagnostics'.
2025-07-14 15:47:55.071 [info] Params: {
    "uri": "file:///Users/weiwang/repos/gitlab-maven/homepage-api/src/homepageapi/repository/content_management/condition.py",
    "diagnostics": []
}
```
</details>

## Outcome When Attempting Debugging Steps:

Did running it from the command line work? Yes.
Running the exact mypy command from the log in the integrated terminal correctly prints the [type-var] error to stdout. The issue is not with mypy itself, but with the extension's handling of the output for this specific type of error.

## Extra Details

- The project uses Poetry for dependency management.
- Mypy configuration is located in pyproject.toml.
- The issue was only discoverable after setting ignore_missing_imports = false, because the library in question was untyped until recently. Now that it is typed, this parsing bug in the extension has become visible.

**Comment by onlywei** [comment]
Hmmm, I got it to work in one repo by removing `non_interactive = true` from my pyproject.toml file.
